### PR TITLE
🔧 DropUpload vue3

### PR DESCRIPTION
**Pull request by roiLeo**
- ref #4750 

<s>⚠️ didn't test</s>

🧪 on `/snek/gallery/3124270296-9`

### CODE REVIEW

File: components/shared/DropUpload.vue [hunk 1]
```diff
@@ -53,107 +53,94 @@
...
-  public fileSizeLimit = MAX_UPLOADED_FILE_SIZE
-  public url = ''
-  public hasError = false
-  public checkFailed = false
-  public fileSizeFailed = false
-  public supportedModelMediaFileExtensions = ['glb']
-  @Ref('upload') readonly upload
+  },
+  icon: { type: String, default: 'upload' },
+  required: { type: Boolean, default: true },
```
Comment (Jarsen136) [inline components/shared/DropUpload.vue:68]: why changing default value from false to true

File: components/shared/DropUpload.vue [hunk 2]
```diff
@@ -53,107 +53,94 @@
...
+    file.value = null
+    return false
+  }
+  fileSizeFailed.value = false
+  checkFailed.value = false
+  const reader = new FileReader()
+  reader.onload = () => {
+    // handleSelection(reader.result)
+    // $consola.log(reader.si);
+  }
```
Comment (Jarsen136) [inline components/shared/DropUpload.vue:129]:
```suggestion
```

File: components/shared/DropUpload.vue (hunk 1 above)
Comment (roiLeo) [inline components/shared/DropUpload.vue:68]:
```suggestion
  required: { type: Boolean, default: false },
```

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
https://deploy-preview-6790--koda-canary.netlify.app/bsx/gallery/5347553-1
drag and drop ✔️ 
image loads ✔️

File: components/shared/DropUpload.vue [hunk 3]
```diff
@@ -53,107 +53,90 @@
...
+  const reader = new FileReader()
+  emit('input', inputFile)
+  if (props.preview) {
+    url.value = URL.createObjectURL(inputFile)
+    hasError.value = false
   }
+  reader.readAsText(inputFile)
 }
+
+onMounted(() => {
```
Comment (vikiival) [inline components/shared/DropUpload.vue:134]: https://vueuse.org/core/useEventListener/

**Review by yangwao — APPROVED** [review]

**Review by vikiival — APPROVED** [review]
.-

**Pull request merged**